### PR TITLE
Updating filter feature in event feed

### LINKED ISSUE #119: Update the filter feature

### PR DESCRIPTION
**Pull request by Sendy459**
### What this PR does?

Updating filter feature in event feed so it will have filtering by two options at the same time 
, option to reset the filtering, and order the events by time or participants number.

### Notes for reviewer
image

### Related Issue

Resolves #119 

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes
- [] The changes in this PR do not require tests
- [x] The changes in this PR  require tests that are not added yet
#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

### CODE REVIEW

File: event/templates/event/all_events.html
Comment (rotem2022) [inline event/templates/event/all_events.html:1]: I think the  `</div>` in line 63 unnecessary and causes the code to be indented incorrectly.

File: event/templates/event/all_events.html [hunk 1]
```diff
@@ -4,31 +4,31 @@
     <button class="EventButton"><a href="{% url 'create_event' user_id=user_id %}">Create Event</a></button>
```
Comment (rotem2022) [inline event/templates/event/all_events.html:4]: I'm not sure if it was intentional, but I noticed you have `user_id=user_id`, while in line 58, it reads `user_id = user.id`. Perhaps you intended for both instances to be `user_id = user.id`?

File: event/views.py [hunk 2]
```diff
@@ -65,19 +65,23 @@ def view_event(request, user_id):
...
+            events = events.filter(location__name=location)
 
-    user_id = request.user.id if request.user.is_authenticated else None
+        order_by = request.GET.get('Order_By', 'None')
+        if order_by == 'Time':
+            events = events.order_by('start_time')
+        elif order_by == 'Participants':
+            events = events.order_by('participants_num')
 
+    user_id = request.user.id if request.user.is_authenticated else None
```
Comment (rotem2022) [inline event/views.py:84]: I noticed that there usage of the `user_id` until this line, where you assign a value to it. 
Therefore, it's possible that the method doesn't actually require `user_id` as an argument, or maybe the assignment is redundant.

**Review by rotem2022 — COMMENTED** [review]
Very good job, I think the changes you add are making the filter feature match better.
left some comments

File: event/views.py (hunk 2 above)
Comment (Sendy459) [reply event/views.py:84]: This value is then included in the context dictionary and passed to the template, user_id parameter is required to ensure that the correct value is passed to the template and used as needed.

File: event/templates/event/all_events.html
Comment (Sendy459) [reply event/templates/event/all_events.html:1]: you are right I forgot to delete it

File: event/templates/event/all_events.html (hunk 1 above)
Comment (Sendy459) [reply event/templates/event/all_events.html:4]: thanks ,both needs to be user_id=user_id as I take them from the context dictionary.

**Review by rotem2022 — DISMISSED** [review]

**Review by SaarOrus — APPROVED** [review]

File: event/views.py [hunk 3]
```diff
@@ -65,19 +65,22 @@ def view_event(request, user_id):
...
-        if 'Choose_filter' in request.GET:
-            filter_type = request.GET.get('Choose_filter')
 
-            if filter_type == 'Category' and 'Choose_Category' in request.GET:
-                category = request.GET.get('Choose_Category')
-                events = events.filter(category__name=category)
-            elif filter_type == 'Location' and 'Choose_Location' in request.GET:
-                location = request.GET.get('Choose_Location')
-                events = events.filter(location__name=location)
+    if 'Choose_Category' in request.GET and request.GET.get('Choose_Category') != 'None':
```
Comment (ygalblum) [inline event/views.py:69]: You should change this to the same way you check `order_by`
```suggestion
    category = request.GET.get('Choose_Category', 'None')
    if category  != 'None':
```

File: event/views.py [hunk 4]
```diff
@@ -65,19 +65,22 @@ def view_event(request, user_id):
...
+        category = request.GET.get('Choose_Category')
+        events = events.filter(category__name=category)
 
-    user_id = request.user.id if request.user.is_authenticated else None
+    if 'Choose_Location' in request.GET and request.GET.get('Choose_Location') != 'None':
```
Comment (ygalblum) [inline event/views.py:73]: Same comment as for `Choose_Category`

File: event/views.py (hunk 2 above)
Comment (ygalblum) [reply event/views.py:83]: So, what value is set in the `user_id` argument? If it's the same as `request.user.id` then you don't need to assign it again. If not, please rename this one to differentiate between the two

**Review by rotem2022 — APPROVED** [review]

**Review by ygalblum — APPROVED** [review]

**Pull request merged**